05/app.js

```diff
@@ -0,0 +1,65 @@
+function Student(initFirstName, initLastName) {
+    this.firstName = initFirstName;
+    this.lastName = initLastName;
+    this.grades = {};
+}
+
+Student.prototype.addGrade = function(subject, grade) {
+    if(typeof this.grades[subject] === 'undefined') {
+        this.grades[subject] = [];
+    }
+    this.grades[subject].push(grade);
+}
+
+Student.prototype.getAverage = function(subject) {
+    if(typeof subject !== 'undefined') {
+        for(const key in this.grades) {
+            let sum = 0;
+            const gradeArr = this.grades[subject];
+            console.log(gradeArr);
+            for(let i = 0; i < gradeArr.length; i++) {
+                sum += gradeArr[i];
+            }
+            return (sum / gradeArr.length).toFixed(2);
+        }
+    } else {
+        const allGrades = [];
+        let sum = 0;
+        for(const item in this.grades) {
```

Comment:
Nie do końca jest to `item`, bardziej `key` - myślę, że ta opcja jest jaśniejsza